watchtower-cli/src/main/java/com/tracelink/appsec/watchtower/cli/executor/UploadScanExecutor.java

```diff
@@ -0,0 +1,244 @@
+package com.tracelink.appsec.watchtower.cli.executor;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.tracelink.appsec.watchtower.cli.reporter.AbstractReporter;
+import com.tracelink.appsec.watchtower.cli.reporter.ConsoleReporter;
+import com.tracelink.appsec.watchtower.cli.reporter.CsvReporter;
+import com.tracelink.appsec.watchtower.cli.scan.ScanStatus;
+import com.tracelink.appsec.watchtower.cli.scan.UploadScanResult;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import kong.unirest.HttpResponse;
+import kong.unirest.Unirest;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles execution of a Watchtower upload scan given a set of parameters for configuration. Zips
+ * target code, uploads to Watchtower for scanning, and reports results to the console or CSV.
+ *
+ * @author mcool
+ */
+public class UploadScanExecutor {
+
+	private static final Logger LOG = LoggerFactory.getLogger(UploadScanExecutor.class);
+	private static final String UPLOAD_SCAN_ENDPOINT = "rest/uploadscan";
+	private static final int NUM_RETRIES = 10;
+	private static final int SLEEP_SECONDS = 6;
+	private static final Gson GSON = new GsonBuilder().registerTypeAdapter(LocalDateTime.class,
+			new LocalDateTimeTypeAdapter()).create();
+
+	private final UploadScanParameters params;
+
+	public UploadScanExecutor(UploadScanParameters params) {
+		this.params = params;
+	}
+
+	/**
+	 * Executes the Watchtower upload scan using the {@link UploadScanParameters} for this class.
+	 *
+	 * @throws IOException          if there is an IO problem with the target or output files
+	 * @throws InterruptedException if there is a problem sleeping while polling for scan results
+	 */
+	public void executeUploadScan() throws IOException, InterruptedException {
+		Path zipDestination = zipTarget(params.getTarget());
+		UploadScanResult result = initiateUploadScan(zipDestination);
+		pollForScanResults(result);
+		if (zipDestination != null) {
+			FileUtils.deleteQuietly(zipDestination.toFile());
+		}
+	}
+
+	/**
+	 * Zips the given target file or directory and returns the path to the zipped file.
+	 *
+	 * @param target the file or directory to zip
+	 * @return path to the zipped file
+	 * @throws IllegalArgumentException if the target directory or file does not exist
+	 * @throws IOException              if there is a problem zipping the target
+	 */
+	private Path zipTarget(String target) throws IllegalArgumentException, IOException {
```

Comment:
watchtower also has a zip4j library already in it, so you could use that here too